File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Model/IfStatement.cs
using Assignement7.Collection;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.Model
{
    public class IfStatement : IStatement
    {
        private Expression expression;
        private IStatement thenStatement;
        private IStatement elseStatement;

        public IfStatement(Expression expression, IStatement thenStatement, IStatement elseStatement)
        {
            this.expression = expression;
            this.thenStatement = thenStatement;
            this.elseStatement = elseStatement;
        }

        public ProgramState Execute(ProgramState state)
        {
            MyIStack<IStatement> stack = state.getExecutionStack();
            int value = 0;
            value = expression.Evaluate(state.getSymbolTable());
            if (value != 0) stack.Push(thenStatement);
            else stack.Push(elseStatement);
            return state;
        }

        public override string ToString()
        {
            return "if(" + expression.ToString() + ") then " + thenStatement.ToString() + " else " + elseStatement.ToString();
        }
    }
}

File: Semester3/Advanced Methods of Programming/Toy Language Interpreter JavaFx/Assignment8/src/Model/Statement/IfStatement.java
package Model.Statement;

import Collection.Stack.MyIStack;
import Model.Exceptions.ToyLanguageInterpreterException;
import Model.Expression.Expression;
import Model.ProgramState;

public class IfStatement implements IStatement{
    private Expression expression;
    private IStatement thenStatement;
    private IStatement elseStatement;

    public IfStatement(Expression expression, IStatement thenStatement, IStatement elseStatement){
        this.expression = expression;
        this.thenStatement = thenStatement;
        this.elseStatement = elseStatement;
    }

    @Override
    public ProgramState execute(ProgramState state) throws ToyLanguageInterpreterException {
        MyIStack<IStatement> stack = state.getExecutionStack();
        int value = 0;
        try{
            value = expression.evaluate(state.getSymbolTable(), state.getHeap());
        }
        catch (Exception e){
            throw new ToyLanguageInterpreterException(e.getMessage());
        }
        if(value != 0) stack.push(thenStatement);
        else stack.push(elseStatement);
        return null;
    }

    @Override
    public String toString(){
        return "if(" + expression.toString() + ") then " + thenStatement.toString() + " else " + elseStatement.toString();
    }
}
